lib/graphiti/rails.rb

```diff
@@ -7,89 +8,25 @@ module Graphiti
-      DefaultExceptionHandler
-    end
-
-    if defined?(Graphiti::Errors::InvalidRequest)
-      register_exception Graphiti::Errors::InvalidRequest,
-        handler: InvalidRequestExceptionHandler
-    end
-
-    # Returns a boolean of whether the specified content type should be handled by a Graphiti exception handler.
-    # The list of format can be found in `.handled_exception_formats`.
-    # @param content_type [Mime::Type]
-    # @return [Boolean]
-    def self.render_exception_for_format?(content_type)
-      handled_exception_formats.include?(content_type.to_sym)
-    end
-
-    # @return [GraphitiErrors::ExceptionHandler]
-    def self.exception_handler_for(exception)
-      # TODO: Move this into GraphitiErrors
-      _errorable_registry[exception.class] || default_exception_handler.new
-    end
-
-    # @param [Exception] exception
-    # @param [Boolean] show_raw_error (false)
-    # @return [Array<Integer, Hash>] HTTP status and payload
-    def self.exception_details(exception, show_raw_error: false)
-      # TODO: Move this into GraphitiErrors
-      exception_klass = exception_handler_for(exception)
-      exception_klass.show_raw_error = show_raw_error
-      status = exception_klass.status_code(exception)
-      payload = exception_klass.error_payload(exception)
-      [status, payload]
-    end
-
-    # @param [Exception] exception
-    # @param [Mime::Type] content_type
-    # @param [Hash] keywords passed through to {.exception_details}
-    # @return [Array<Integer, Mime::Type, String>] HTTP status, content type, and payload
-    def self.rendered_exception(exception, content_type:, **keywords)
-      status, body = exception_details(exception, **keywords)
+    autoload :ExceptionHandler, "graphiti/rails/exception_handlers"
+    autoload :InvalidRequestHandler, "graphiti/rails/exception_handlers"
 
-      to_format = (content_type == :jsonapi) ? "to_json" : "to_#{content_type.to_sym}"
-
-      if content_type && body.respond_to?(to_format)
-        formatted_body = body.public_send(to_format)
-        format = content_type
-      else
-        formatted_body = body.to_json
-        format = Mime[:json]
-      end
-
-      [status, format, formatted_body]
+    def self.included(_klass)
+      ActiveSupport::Deprecation.warn("With graphiti-rails, including Graphiti::Rails is unnecessary")
     end
   end
 end
 
 ActiveSupport.on_load(:action_controller) do
   include Graphiti::Rails::Context
   include Graphiti::Rails::Debugging
+
+  register_exception Graphiti::Errors::InvalidRequest,   status: 400, handler: Graphiti::Rails::InvalidRequestHandler
+  register_exception Graphiti::Errors::RecordNotFound,   status: 404, handler: Graphiti::Rails::ExceptionHandler
+  register_exception Graphiti::Errors::RemoteWrite,      status: 400, handler: Graphiti::Rails::ExceptionHandler
+  register_exception Graphiti::Errors::SingularSideload, status: 400, handler: Graphiti::Rails::ExceptionHandler
```

Comment:
This is perfect, exactly what I was hoping to see. Should we maybe default the handler to `Graphiti::Rails::ExceptionHandler`?

Reply:
I was thinking about being able to set a default as well as allowing for them to be set as symbols as a shorthand

Reply:
https://github.com/wagenet/rescue_registry/issues/2